mysite/blog/views.py

```diff
@@ -59,11 +62,26 @@ def Profileview(request,name):
 
 class PostDetailView(DetailView):
     model = Post
+    def get_object(self):
+        obj = super().get_object()
+        obj.view_count += 1
+        obj.save()
+        return obj
+
+    # def get_context_data(self, *args, **kwargs):
+    #     context = super().get_context_data(**kwargs)
+    #     texts = self.object.content
+    #     print(texts)
```

Comment:
@ManishShah120 what outputs you are getting in here? I mean did you figure out which part of this function is actually not working as expected?

Reply:
```
    def get_context_data(self, *args, **kwargs):
        context = super().get_context_data(**kwargs)
        texts = self.object.content
        #print(texts)
        Read_Time=get_read_time(texts)
        print(type(Read_Time))
        print(Read_Time)
        #Post.objects.create(read_time=Read_Time)
        return context
```
By Uncommenting these lines this is the output I am getting:-
```
<class 'str'>
0:01:00
[04/Sep/2020 11:31:12] "GET /post/lorem-ipsum/ HTTP/1.1" 200 4665
```

Now I am trying to save `0:01:00` to the database, which is creating the error

---

When I am trying to save this value to the database:-
```
    def get_context_data(self, *args, **kwargs):
        context = super().get_context_data(**kwargs)
        texts = self.object.content
        #print(texts)
        Read_Time=get_read_time(texts)
        print(type(Read_Time))
        print(Read_Time)
        Post.objects.create(read_time=Read_Time)
        return context
```
This is error message I am getting:-
```
Environment:


Request Method: GET
Request URL: http://127.0.0.1:8000/post/lorem-ipsum/

Django Version: 3.0.8
Python Version: 3.8.2
Installed Applications:
['blog.apps.BlogConfig',
 'django.contrib.admin',
 'crispy_forms',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'users.apps.UsersConfig']
Installed Middleware:
('whitenoise.middleware.WhiteNoiseMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware')



Traceback (most recent call last):
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/backends/utils.py", line 86, in _execute
    return self.cursor.execute(sql, params)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/backends/sqlite3/base.py", line 396, in execute
    return Database.Cursor.execute(self, query, params)

The above exception (NOT NULL constraint failed: blog_post.author_id) was the direct cause of the following exception:
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/views/generic/base.py", line 71, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/views/generic/base.py", line 97, in dispatch
    return handler(request, *args, **kwargs)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/views/generic/detail.py", line 107, in get
    context = self.get_context_data(object=self.object)
  File "/home/manish/Videos/open-source/Blogger/mysite/blog/views.py", line 78, in get_context_data
    Post.objects.create(read_time=Read_Time)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/models/query.py", line 433, in create
    obj.save(force_insert=True, using=self.db)
  File "/home/manish/Videos/open-source/Blogger/mysite/blog/models.py", line 22, in save
    super(Post, self).save(*args, **kwargs)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/models/base.py", line 745, in save
    self.save_base(using=using, force_insert=force_insert,
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/models/base.py", line 782, in save_base
    updated = self._save_table(
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/models/base.py", line 887, in _save_table
    results = self._do_insert(cls._base_manager, using, fields, returning_fields, raw)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/models/base.py", line 924, in _do_insert
    return manager._insert(
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/models/query.py", line 1204, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/models/sql/compiler.py", line 1392, in execute_sql
    cursor.execute(sql, params)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/backends/utils.py", line 100, in execute
    return super().execute(sql, params)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/backends/utils.py", line 68, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/backends/utils.py", line 77, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/backends/utils.py", line 86, in _execute
    return self.cursor.execute(sql, params)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/backends/utils.py", line 86, in _execute
    return self.cursor.execute(sql, params)
  File "/home/manish/Videos/open-source/Blog-Env/lib/python3.8/site-packages/django/db/backends/sqlite3/base.py", line 396, in execute
    return Database.Cursor.execute(self, query, params)

Exception Type: IntegrityError at /post/lorem-ipsum/
Exception Value: NOT NULL constraint failed: blog_post.author_id
```

Reply:
@ManishShah120 there is no need to store the expected view time in db. It's most of a frontend work. So, you just fetch the blog data from backend, calculate the expected read time and display it on frontend, no need to store it in db as we are not going to use it anywhere else. Can you please try this work flow, it should work then.

Reply:
Sure.